hypothesis-python/src/hypothesis/core.py

```diff
@@ -931,7 +933,18 @@ def wrapped_test(*arguments, **kwargs):
                             report(
                                 'You can add @seed(%d) to this test to '
                                 'reproduce this failure.' % (generated_seed,))
-                raise
+                if PY2:
```

Comment:
In general I have a *very* strong preference against `if PY2` code outside of `compat.py`. Can this not be refactored to a function?

Reply:
Not without adding another stack frame, and we've just gone to a fair bit of trouble to get rid of those!

Otherwise it absolutely would be `compat.raise_with_traceback(error, tb)`.

Reply:
Fair enough. Could you add an explanatory comment to the code then, please?

(In general I think this section could use some fairly verbose commenting because even knowing what it's supposed to do I'm finding it a bit eyebrow raising)

Reply:
Yep, comments incoming.  I'll also try to refactor things a bit so the logic is as clear as possible without comments too!